sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -98,9 +98,10 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
    * }}}
    */
   override def visitAnalyze(ctx: AnalyzeContext): LogicalPlan = withOrigin(ctx) {
-    if (ctx.partitionSpec == null &&
-      ctx.identifier != null &&
-      ctx.identifier.getText.toLowerCase == "noscan") {
+    if (ctx.partitionSpec == null && ctx.identifier != null) {
```

Comment:
What if partition spec is not null ?
What happens with something like 
```
ANALYZE TABLE mytable PARTITION (a) garbage
```
(Could you add a test for that ?)
Maybe 
```
if (ctx.identifier != null && ctx.identifier.getText.toLowerCase != "noscan") {
  throw new ParseException(s"Expected `NOSCAN` instead of `${ctx.identifier.getText}`", ctx)
}
```
could be moved to the top ?